src/main/scala/satify/model/problems/Problem.scala

```diff
@@ -2,10 +2,24 @@ package satify.model.problems
 
 import satify.model.expression.Expression
 
+import scala.swing.Component
+
+/** Entity representing a SAT problem. */
 trait Problem:
+  /** The expression encoded of the problem that must be satisfied. */
   val exp: Expression
 
-enum ProblemChoice:
-  case GraphColoring
-  case NQueens
-  case NurseScheduling
+  /** The set of problem's constraints, with a description. */
+  val constraints: Set[(String, Expression)]
```

Comment:
Where the description is used? If it is not useful I would delete it, and add some doc to clarify the constraint.

Reply:
I want to give importance to the constraints, independently from the expression. It is easier understand constraints from a set providing a description, rather than from directly in an expression.